app/Repositories/BuildRepository.php

```diff
@@ -157,15 +155,34 @@ public function getLastBuild(Application $application, Platform $platform)
      * Return the build grouped by platform
      *
      * @param  Application  $application
+     * @param  int $applicationActiveDevices
      * @return Collection
      */
-    public function getByPlatform(Application $application): Collection
+    public function getByPlatform(Application $application, int $applicationActiveDevices): Collection
     {
         return $application->builds()
             ->latest()
             ->get()
             ->groupBy('platform')
-            ->sortKeys();
+            ->sortKeys()
+            ->map(function (Collection $platformCollection) use ($applicationActiveDevices) {
+                return $platformCollection->map(function (Build $build) use ($applicationActiveDevices) {
+
+                    // clear 'appends', else `installations` and `downloads` will be called by the resource
+                    $build->setAppends([]);
+
+                    if (! $build->partial_rollout && $build->rollout_percentage !== 100) {
+                        $build->rollout_percentage = 100;
+                    }
+
+                    $build->rollout_installations = $build->installations;
```

Comment:
What about dedicated resource instead setting dynamic properties?

Reply:
I thought about it too. The idea is to avoid having a business logic in the resource layer.